gdpr/vendorlist-fetching_test.go

```diff
@@ -7,233 +7,697 @@ import (
-	assertNilErr(t, err)
-	vendor := list.Vendor(32)
-	assertBoolsEqual(t, true, vendor.Purpose(1))
-	assertBoolsEqual(t, false, vendor.Purpose(3))
-	assertBoolsEqual(t, false, vendor.Purpose(4))
-
-	list, err = fetcher(context.Background(), 2)
-	assertNilErr(t, err)
-	vendor = list.Vendor(32)
-	assertBoolsEqual(t, true, vendor.Purpose(1))
-	assertBoolsEqual(t, true, vendor.Purpose(3))
-}
-
-func TestLazyFetch(t *testing.T) {
-	firstVendorList := mockVendorListData(t, 1, map[uint16]*purposes{
-		32: {
-			purposes: []int{1, 2},
-		},
-	})
-	secondVendorList := mockVendorListData(t, 2, map[uint16]*purposes{
-		3: {
-			purposes: []int{1},
-		},
-	})
-	server := httptest.NewServer(http.HandlerFunc(mockServer(1, map[int]string{
-		1: firstVendorList,
-		2: secondVendorList,
+	testCases := []test{
+		{
+			description: "Fetch - No Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: false,
+				vendorListVersion:  1,
+			},
+			expected: vendorList1Expected,
+		},
+		{
+			description: "Fetch - No Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: false,
+				vendorListVersion:  2,
+			},
+			expected: vendorList2Expected,
+		},
+		{
+			description: "Fetch - Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: true,
+				vendorListVersion:  1,
+			},
+			expected: vendorList1Expected,
+		},
+		{
+			description: "Fetch - Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: true,
+				vendorListVersion:  2,
+			},
+			expected: vendorList2Expected,
+		},
+		{
+			description: "No Fetch - Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: true,
+				vendorListVersion:  1,
+			},
+			expected: vendorListFallbackExpected,
+		},
+		{
+			description: "No Fetch - Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: true,
+				vendorListVersion:  2,
+			},
+			expected: vendorListFallbackExpected,
+		},
+		{
+			description: "No Fetch - No Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: false,
+				vendorListVersion:  1,
+			},
+			expected: testExpected{
+				errorMessage: "gdpr vendor list version 1 does not exist, or has not been loaded yet. Try again in a few minutes",
+			},
+		},
+		{
+			description: "No Fetch - No Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: false,
+				vendorListVersion:  2,
+			},
+			expected: testExpected{
+				errorMessage: "gdpr vendor list version 2 does not exist, or has not been loaded yet. Try again in a few minutes",
+			},
+		},
+	}
+
+	for _, test := range testCases {
+		runTest(t, test, tcf1Version, server)
+	}
+}
+
+func TestTCF2FetcherInitialLoad(t *testing.T) {
```

Comment:
Added TCF2 tests. In the future, I figure we'll just delete the TCF1 stuff when we deprecate support.